_test/test_utilities/test_deterministic_pseudorandom_sequence.m

```diff
@@ -0,0 +1,68 @@
+classdef test_deterministic_pseudorandom_sequence < TestCase
+%=========================================================================
+% Test(s) to ensure that the "pseudo-random" sequence generated by
+% the noisify_test_rand class over multiple calls to one of its objects
+% for the myrand method yield the same sequence as one created directly
+% over an explicit single range.
+%
+% Cloned from "test_rand_like.m" without altering that script's 
+% particular idiosyncrasies.
+%=========================================================================
+
+    properties
+        % Current path for noisify_test_rand.
+        % Needs to be added and removed from the path to make the
+        % test function
+        deterministic_pseudorandom_sequence_path = '..\shared';
+    end
+    methods
+        
+        % constructor for the test class - constructs base
+        % TestClass instance and minimally displays the name argument.
+        function this=test_deterministic_pseudorandom_sequence(name)
+            this = this@TestCase(name);
+            disp(name)
+        end
+        
+        % test function to verify operation of noisify_test_rand
+        function test_if_deterministic_pseudorandom_sequence_is_ok(this)
+            % find noisify_test_rand
+            addpath(this.deterministic_pseudorandom_sequence_path);
+            % create object for deterministic_psuedorandom_sequence to
+            % initialise state at the start of the deterministic
+            % sequence and hold the state for multiple calls.
+            my_sequence_gen = deterministic_pseudorandom_sequence();
+            % create one sequence of size sy1
+            % NB sizes are designed to push the sequence through 
+            %    the value 999 where the sequence restarts from 0
+            sy1 = 1191;
+            dy1 = my_sequence_gen.myrand([1,sy1]);
+            % create a second sequence of the same size, starting
+            % after the endpoint of the first sequence
+            sy2 = 1191;
+            dy2 = my_sequence_gen.myrand([1,sy2]);
```

Comment:
This test seems to be testing several things. 

It looks like you're testing that:

1. Two sequences of the same size are continued on consecutive calls
2. The sequence generator can be reset
3. Two sequences of different sizes are continued on consecutive calls

I'd have these as separate tests if that's possible.